lib/sparse/src/index/migrate.rs

```diff
@@ -0,0 +1,50 @@
+use std::path::Path;
+
+use io::file_operations::{FileOperationResult, FileStorageError};
+use io::storage_version::StorageVersion;
+
+pub struct SparseVectorIndexVersion;
+
+impl StorageVersion for SparseVectorIndexVersion {
+    fn current_raw() -> &'static str {
+        env!("CARGO_PKG_VERSION")
+    }
+}
+
+pub fn migrate(path: &Path) -> FileOperationResult<()> {
+    let app_version = SparseVectorIndexVersion::current();
+    match SparseVectorIndexVersion::load(path)? {
+        Some(version) if version == app_version => (),
+        Some(version) => {
+            return Err(FileStorageError::generic(format!(
+                "Version mismatch: expected {app_version}, found {version}"
+            )))
+        }
+        None => migrate_from_v1(path)?,
+    }
+    Ok(())
+}
+
+fn migrate_from_v1(path: &Path) -> FileOperationResult<()> {
+    // Disable migration for now.
+    SparseVectorIndexVersion::save(path)?;
```

Comment:
In the current state of this PR, only a file named `version.info` with contents `0.1.0` is created. No file with a `v2` in its name is created.

To enable the actual migration, we'll revert the last commit of this PR.

Reply:
Out of curiosity, can't we reuse the same file path and rely on the surrounding version number of contents to tell us whether it has been migrated. Or do you deem that as not reliable enough?

Reply:
> No file with a v2 in its name is created.

Yes, but once implemented, the next migration might possibly see either no version file or `0.1.0`. 

I am ok with this approach, but please keep in mind that it is possible, that we will release another patch version in between your PRs

Reply:
> can't we reuse the same file path and rely on the surrounding version

It might be potentially corruptive, if the process will be killed in between the migration.